### PR TITLE
PEPPER-1188 Third time's the charm with redisson upgrade.

### PR DESCRIPTION
**Pull request by zyme**
PEPPER-1188.  Updated CacheServiceTest test to better isolate the issues that showed up on dev, and then started playing version whackamole and discovered that the redisson 3.19.1 -> 3.19.2 version change is where the failure was introduced.  

Traced it to our use of an incompatible version of netty-common, that lacked the NetUtil.NETWORK_INTERFACES field.  [Previous version without the field here](https://github.com/netty/netty/blob/netty-4.1.50.Final/common/src/main/java/io/netty/util/NetUtil.java), more recent [version that this PR uses here](https://github.com/netty/netty/blob/netty-4.1.100.Final/common/src/main/java/io/netty/util/NetUtil.java). Once I upgraded to a version of netty-common that had the NetUtil.NETWORK_INTERFACES field, things started to work again.

Along the way, while I was trying out some other potential fixes, I centralized the jedis version (although it has not changed).

## FUD Score

_Overall, how are you feeling about these changes?_

- [ x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have added regression test labels to the jira ticket
- [ ] I have added verification steps to the jira ticket
- [x ] I have written automated positive tests
- [ x] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x ] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/dss-core/pom.xml [hunk 1]
```diff
@@ -394,15 +394,12 @@
         <dependency>
             <groupId>redis.clients</groupId>
             <artifactId>jedis</artifactId>
+            <version>${jedis.version}</version>
```
Comment (zyme) [inline pepper-apis/dss-core/pom.xml:397]: No change to version, but put it in a central place for better future management.

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/cache/CacheService.java [hunk 2]
```diff
@@ -170,7 +170,7 @@ public void resetAllCaches() {
         }
         if (!(cacheManager instanceof NullCacheManager)) {
             redissonClient.getKeys().getKeysByPattern(LOCAL_CACHE_PREFIX + "*").forEach(cacheKey -> {
-                redissonClient.getLocalCachedMap(cacheKey, new FstCodec(), LocalCachedMapOptions.defaults()).delete();
+                redissonClient.getLocalCachedMap(cacheKey, new Kryo5Codec(), LocalCachedMapOptions.defaults()).delete();
```
Comment (zyme) [inline pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/cache/CacheService.java:173]: Previous PRs were missing this explicit use of the most up to date codec.

File: pepper-apis/dss-core/src/test/java/org/broadinstitute/ddp/cache/CacheServiceTest.java [hunk 3]
```diff
@@ -13,25 +14,14 @@ public class CacheServiceTest {
...
-        } catch (NoSuchMethodError e) {
+        } catch (RedisConnectionException e) {
+            log.info("As expected, cannot connect to local instance that does not exist.");
+        } catch (NoSuchFieldError | NoSuchMethodError e) {
             e.printStackTrace();
-            Assert.fail("Transitive dependency error with redisson: " + e.getMessage());
-        } catch (NoSuchFieldError e) {
             Assert.fail("Could not get CacheService instance: " + e.getMessage());
         } catch (Throwable e) {
-            boolean hasConfigFileError = false;
```
Comment (zyme) [inline pepper-apis/dss-core/src/test/java/org/broadinstitute/ddp/cache/CacheServiceTest.java:22]: Now that this same issue was showing up in the full dev environment, it became clear that the way I was trying to isolate and ignore the fact that the test couldn't connect to localhost was actually obscuring the real problem.

File: pepper-apis/parent-pom.xml [hunk 4]
```diff
@@ -469,17 +469,12 @@
...
             <artifactId>jedis</artifactId>
-            <version>3.3.0</version>
+            <version>${jedis.version}</version>
         </dependency>
         <dependency>
             <groupId>io.netty</groupId>
             <artifactId>netty-common</artifactId>
-            <version>4.1.50.Final</version>
-        </dependency>
-        <dependency>
```
Comment (zyme) [inline pepper-apis/parent-pom.xml:479]: No longer need this dependency since we are using Kryo5Codec.

File: pepper-apis/parent-pom.xml [hunk 5]
```diff
@@ -469,17 +469,12 @@
         <dependency>
             <groupId>redis.clients</groupId>
             <artifactId>jedis</artifactId>
-            <version>3.3.0</version>
+            <version>${jedis.version}</version>
         </dependency>
         <dependency>
             <groupId>io.netty</groupId>
             <artifactId>netty-common</artifactId>
-            <version>4.1.50.Final</version>
```
Comment (zyme) [inline pepper-apis/parent-pom.xml:477]: Moving to `4.1.100.Final`, now in a centralized `netty.version`.

**Review by ssettipalli — APPROVED** [review]
looks good to me

**Pull request merged**